compiler/rustc_interface/src/util.rs

```diff
@@ -559,7 +559,7 @@ pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<C
     // command line, then reuse the empty `base` Vec to hold the types that
     // will be found in crate attributes.
     // JUSTIFICATION: before wrapper fn is available
-    #[cfg_attr(not(bootstrap), allow(rustc::bad_opt_access))]
+    #[allow(rustc::bad_opt_access)]
```

Comment:
this lint seems weird to me - I would expect this to use normal pub/private fields rather than introducing a whole new lint. but not related to your change.

Reply:
see #99710 - sometimes we need the option fields to be public so that they are accessible by `Session` or `TyCtxt` or wherever a wrapper function makes sense to be defined for it, but we also want to prevent it being used directly (preferring the wrapper function)